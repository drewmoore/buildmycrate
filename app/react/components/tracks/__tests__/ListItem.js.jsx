/* globals describe test expect */
import React          from 'react';
import { Provider }   from 'react-redux';
import renderer       from 'react-test-renderer';
import { mount }      from 'enzyme';
import TracksListItem from '../ListItem.js.jsx';
import store          from '../../../__tests__/helpers/store.js.es6';
import '../../../__tests__/helpers/initialize.js.es6';

describe('TracksListItem', () => {
  const props = {
    id:           0,
    title:        'Forty-Six and Two',
    user:         { username: 'Tool' },
    bpm:          80,
    keySignature: 'dmin',
    purchaseUrl:  'https://www.amazon.com/gp/search?ie=UTF8&keywords=Tool',
    streamUrl:    'https://www.last.fm/music/Tool/_/Forty+Six+&+2',
    downloadable: true,
    purchaseable: true,
    streamable:   true,
    turntables:   [{ id: 0 }, { id: 1 }],
    download:     () => {}
  };
  const template = (<Provider store={store}><TracksListItem {...props} /></Provider>);

  describe('DOM Snapshot', () => {
    const tree = renderer.create(template).toJSON();

    test('should match snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Rendering Props', () => {
    const listItem  = mount(template);

    test('displays title', () => {
      expect(listItem.html()).toMatch(props.title);
    });

    test('displays user.username', () => {
      expect(listItem.html()).toMatch(props.user.username);
    });

    test('displays bpm', () => {
      expect(listItem.html()).toMatch(props.bpm.toString());
    });

    test('displays keySignature', () => {
      expect(listItem.html()).toMatch(props.keySignature);
    });

    test('displays puchase link', () => {
      expect(listItem.find('.purchase-link').getDOMNode().href).toMatch(props.purchaseUrl);
    });

    test('renders buttons for streaming track', () => {
      expect(listItem.find('[data-hook="turntable-icon"]')).toHaveLength(props.turntables.length);
    });
  });
});
