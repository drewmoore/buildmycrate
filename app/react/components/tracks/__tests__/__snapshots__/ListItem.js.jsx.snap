// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TracksListItem DOM Snapshot should match snapshot 1`] = `
<div
  className="row track"
>
  <div
    className="col-xs-3 col-lg-4"
  >
    Forty-Six and Two
  </div>
  <div
    className="col-xs-3 col-lg-4"
  >
    Tool
  </div>
  <div
    className="col-xs-3 col-lg-1"
  >
    80
  </div>
  <div
    className="col-xs-3 col-lg-1"
  >
    dmin
  </div>
  <div
    className="col-xs-3 col-lg-2 text-right"
  >
    <img
      alt="Download Track Button"
      className="track-button clickable"
      onClick={[Function]}
      src={undefined}
    />
    <a
      className="purchase-link"
      href="https://www.amazon.com/gp/search?ie=UTF8&keywords=Tool"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Purchase Track Button"
        className="track-button clickable"
        src={undefined}
      />
    </a>
    <span>
      <img
        alt="Button to load left turntable"
        className="track-button clickable"
        data-hook="turntable-icon"
        onClick={[Function]}
        src="/turntable-left.jpg"
      />
      <img
        alt="Button to load right turntable"
        className="track-button clickable"
        data-hook="turntable-icon"
        onClick={[Function]}
        src="/turntable-right.jpg"
      />
    </span>
  </div>
</div>
`;
